Xamarin.PropertyEditing.Tests/SimpleCollectionViewTests.cs

```diff
@@ -0,0 +1,652 @@
+			Assert.That (view, Does.Not.Contain (nodes[0]));
+			Assert.That (view, Contains.Item (nodes[1]));
+		}
+
+		[Test]
+		public void FilterGroupedIndexes()
+		{
+			TestNode[] nodes = new TestNode[] {
+				new TestNode ("A"),
+				new TestNode ("Ab") { Flag = true },
+				new TestNode ("Ac") { Flag = true },
+				new TestNode ("Ad"),
+				new TestNode ("Ae") { Flag = true },
+				new TestNode ("Af") { Flag = true }
+			};
+
+			var view = new SimpleCollectionView (nodes, new SimpleCollectionViewOptions {
+				DisplaySelector = TestNodeDisplaySelector
+			});
+
+			Assume.That (view.Cast<object>().Count(), Is.EqualTo (6));
+
+			List<NotifyCollectionChangedEventArgs> args = new List<NotifyCollectionChangedEventArgs>();
+			view.CollectionChanged += (sender, e) => {
+				args.Add (e);
+			};
+
+			view.Options.Filter = o => !((TestNode)o).Flag;
+
+			Assert.That (view, Contains.Item (nodes[0]));
+			Assert.That (view, Contains.Item (nodes[3]));
+			Assert.That (view.Cast<object>().Count(), Is.EqualTo (2));
+
+			Assert.That (args.Count, Is.EqualTo (2));
+			Assert.That (args[0].OldItems, Contains.Item (nodes[1]));
+			Assert.That (args[0].OldItems, Contains.Item (nodes[2]));
+			Assert.That (args[0].OldStartingIndex, Is.EqualTo (1));
+			Assert.That (args[1].OldItems, Contains.Item (nodes[4]));
+			Assert.That (args[1].OldItems, Contains.Item (nodes[5]));
+			Assert.That (args[1].OldStartingIndex, Is.EqualTo (2));
+		}
+
+		[Test]
+		public void InOrder()
+		{
+			TestNode[] nodes = new TestNode[] {
+				new TestNode { Key = "Z" },
+				new TestNode { Key = "C" },
+				new TestNode { Key = "E" },
+				new TestNode { Key = "B" },
+				new TestNode { Key = "A" }
+			};
+
+			var view = new SimpleCollectionView (nodes, new SimpleCollectionViewOptions {
+				DisplaySelector = TestNodeDisplaySelector
+			});
+
+			Assume.That (view, Contains.Item (nodes[0]));
+			Assume.That (view, Contains.Item (nodes[1]));
+			Assume.That (view, Contains.Item (nodes[2]));
+			Assume.That (view, Contains.Item (nodes[3]));
+			Assume.That (view, Contains.Item (nodes[4]));
+
+			Assert.That (view.Cast<object>().ElementAt (0), Is.EqualTo (nodes[4]));
+			Assert.That (view.Cast<object>().ElementAt (1), Is.EqualTo (nodes[3]));
+			Assert.That (view.Cast<object>().ElementAt (2), Is.EqualTo (nodes[1]));
+			Assert.That (view.Cast<object>().ElementAt (3), Is.EqualTo (nodes[2]));
+			Assert.That (view.Cast<object>().ElementAt (4), Is.EqualTo (nodes[0]));
+		}
+
+		[Test]
+		public void AddedInOrder()
+		{
+			var nodes = new ObservableCollection<TestNode> {
+				new TestNode { Key = "Z" },
+				new TestNode { Key = "C" },
+				// F
+				new TestNode { Key = "E" },
+				new TestNode { Key = "B" },
+				new TestNode { Key = "A" }
+			};
+
+			var view = new SimpleCollectionView (nodes, new SimpleCollectionViewOptions {
+				DisplaySelector = TestNodeDisplaySelector
+			});
+
+			Assume.That (view.Cast<object>().ElementAt (0), Is.EqualTo (nodes[4]));
```

Comment:
You do, there's no built-in non-generic version of it. We could add our own which might make this nicer to write going forward, but since this is just for tests it works.

Reply:
Sure, that's probably fine, but note that the code would also be clearer from it. Looks like a side-effect the way it is now.